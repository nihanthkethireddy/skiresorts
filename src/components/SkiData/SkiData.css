.card {
  margin: 10px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.cam {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.inputElements {
  display: flex;
  justify-content: space-between;
  width: 100%;
}

form {
    display: flex;
    flex-direction: column;
    align-items: center;
}