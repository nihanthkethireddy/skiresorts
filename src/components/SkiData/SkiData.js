import { useState } from 'react';
import uuid from 'react-uuid';
import Webcam from 'react-webcam';
import './SkiData.css';

function SkiData(props) {
  const { resort, updateResort, readOnly, setPopup } = props
  const [formData, setformData] = useState({
    id: resort.id || uuid(),
    img: resort.img || '',
    name: resort.name || '',
    runs: resort.runs || ''
  })

  const [errors, setErrors] = useState([])

  const [enableErrors, setEnableErrors] = useState(false)

  const handleUserInput = (e) => {
    const name = e.target.name;
    const value = e.target.value;

    const tempData = { ...formData }
    tempData[name] = value
    setformData(tempData)
    validateField(name, value)
  }

  const WebcamComponent = () => {
    return <Webcam
      audio={false}
      screenshotFormat="image/jpeg"
    >
      {({ getScreenshot }) => (
        <button
          name="img"
          onClick={(e) => {
            e.preventDefault()
            const imageSrc = getScreenshot()
            e.target.value = imageSrc
            handleUserInput(e)
          }}
        >
          Capture photo
        </button>
      )}
    </Webcam>;
  }
    const formSubmit = (e) => {
      e.preventDefault();

      if (errors.length > 0) {
        setEnableErrors(true)
      } else {
        setEnableErrors(false)
        updateResort(formData)
        setPopup({ open: false, resort: {}, readOnly: true })
      }
    }

    const validateField = (fieldName, value) => {
      let tempErrors = [];
      switch (fieldName) {
        case 'name':
          if (value.length == 0) {
            tempErrors.push('Name is required');
          }
          break;
        case 'runs':
          if (!(/^\d+$/.test(value))) {
            tempErrors.push('Runs should contain only Numbers');
          }
          break;
        default:
          break;
      }
      console.log(tempErrors, 'ERR')
      setErrors(tempErrors);
    }

    return (
      <form className="demoForm" onSubmit={e => formSubmit(e)}>
        <div className="panel panel-default">
          {enableErrors ? errors.map(err => <p>{err}</p>) : null}
        </div>
        { !readOnly ? <div className="cam">{WebcamComponent()}</div> : <img src={formData.img}></img> }
        <div>
          <label htmlFor="name">Name</label>
          <input type="text" required className="form-control" name="name"
            placeholder="Resort Name"
            value={formData.name}
            disabled={readOnly}
            onChange={handleUserInput} />
        </div>
        <div>
          <label htmlFor="runs">Runs</label>
          <input type="text" required className="form-control" name="runs"
            placeholder="Ski Runs"
            value={formData.runs}
            disabled={readOnly}
            onChange={handleUserInput} />
        </div>
        <button type="submit" className="btn btn-primary">Save</button>
      </form>
    )
  }

  export { SkiData };
