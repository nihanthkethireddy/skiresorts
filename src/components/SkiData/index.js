import {SkiData} from './SkiData'

export {SkiData}