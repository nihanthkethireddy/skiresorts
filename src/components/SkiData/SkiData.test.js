import { shallow } from 'enzyme';
import { SkiData } from './SkiData';

const props = {
  updateResort: jest.fn(),
  readOnly: false,
  setPopup: jest.fn(),
  resort: {
    id: 1,
    img: null,
    name: 'Resort 1',
    runs: 100
  }
}
describe('<SkiData /> with ReadOnly View', () => {
  const container = shallow(<SkiData {...props} />);
  console.log(container.html(),'+++++++++++++++++++++++++==')
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  it('should contain the webcam component', () => {
    expect(container.find('div.cam').length).toEqual(1);
  });

  it('should contain Save button', () => {
    expect(container.find('button[type="submit"]').length).toEqual(1);
  });


  // it('Check if Edit and Delete buttons are rendered in SkiCard', () => {
  //   expect(container.find('button').length).toEqual(2);
  // });

  // it('Check if the image tag is rendered in SkiCard', () => {
  //   expect(container.find('img').length).toEqual(1);
  // });

});