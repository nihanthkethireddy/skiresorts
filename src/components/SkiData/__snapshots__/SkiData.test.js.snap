// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkiData /> with ReadOnly View should match the snapshot 1`] = `"<form class=\\"demoForm\\"><div class=\\"panel panel-default\\"></div><div class=\\"cam\\"><video autoplay=\\"\\" muted=\\"\\" playsinline=\\"\\"></video><button name=\\"img\\">Capture photo</button></div><div><label for=\\"name\\">Name</label><input type=\\"text\\" required=\\"\\" class=\\"form-control\\" name=\\"name\\" placeholder=\\"Resort Name\\" value=\\"Resort 1\\"/></div><div><label for=\\"runs\\">Runs</label><input type=\\"text\\" required=\\"\\" class=\\"form-control\\" name=\\"runs\\" placeholder=\\"Ski Runs\\" value=\\"100\\"/></div><button type=\\"submit\\" class=\\"btn btn-primary\\">Save</button></form>"`;
